### PR TITLE
Allow other nasm inputformat, add readelf to assembly

### LINKED ISSUE #1804: Default args restrict NASM usage
I want to do assembly for Mach-O 64, but I cannot change the default args to NASM, which specify “-F stabs”.  That is not compatible with “-f macho64.” Specifying the -f option overrides the default ELF option, but there is no way to override the “-F stabs” option when targeting Mach-O 64.
6A4973D6-3C23-4621-9BFF-FC864A0C2C23

### LINKED ISSUE #2221: [REQUEST] nasm fails when compiling with -fbin
**Describe the bug**
When compiling a binary with `-fbin` to just get the binary output nasm fails because it's incompatible with the `stabs` debug mode.

**To Reproduce**
- Create a new assembly file
- add `-fbin` to compile flags

**Expected behavior**
When either selecting compile to binary or `-fbin` are selected flags for setting debug mode should be removed.

**Link**
https://godbolt.org/z/qPWbfj

### PR DESCRIPTION
**Pull request by partouf**
Fixes #2221

Doesn't show assembly for non-elf binaries because objdump does not seem to support it, but it compiles and I've added readelf to see at least "something" about the contents.

Also fixes #1804

### CODE REVIEW

File: lib/compilers/nasm.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2021, Compiler Explorer Authors
```
Comment (mattgodbolt) [inline lib/compilers/nasm.js:1]: 2021! How did that happen?! :)

**Review by mattgodbolt — APPROVED** [review]
Looks good! thanks! Super minor nitpick only :)

File: lib/compilers/assembly.js [hunk 2]
```diff
@@ -69,23 +87,8 @@ export class AssemblyCompiler extends BaseCompiler {
...
-                    }
-                });
-                reject('No output file was generated');
-            });
-        });
-    }
-
     async objdump(outputFilename, result, maxSize, intelAsm, demangle) {
-        const realOutputFilename = await this.getGeneratedOutputFilename(outputFilename);
+        const realOutputFilename = this.getGeneratedOutputFilename(outputFilename);
```
[the remark this answers is not in the document]
Comment (partouf) [reply lib/compilers/assembly.js:91]: I might take another look at the invocations of getOutputFilename(), but the iterations I did before writing an override for getOutputFilename did not allow for async without rewriting the rest of the chain of functions and I didn't want to jump in that deep dark spiral just yet.

**Review by RubenRBS — APPROVED** [review]

File: lib/compilers/assembly.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (partouf) [reply lib/compilers/assembly.js:91]: Only 2 callsites in base-compiler would have to change to async, but getOutputFilename() in basically every other compiler override and a unittest.

It doesn't look impossible, though. But there are too many compilers for me to make the changes confidently.

**Comment by partouf** [comment]
Now live

**Pull request merged**